articles/understanding-map-reduce-in-hadoop/index.md

```diff
@@ -0,0 +1,118 @@
+MapReduce is a component of the Apache Hadoop ecosystem, a framework that enhances massive data processing. Other components of Apache Hadoop include Hadoop Distributed File System (HDFS), Yarn, and Apache Pig. 
+
+The MapReduce component enhances the processing of massive data using dispersed and parallel algorithms in the Hadoop ecosystem. This programming model is applied in social platforms and e-commerce to analyze huge data collected from online users.
+
+This article provides an understanding of MapReduce in Hadoop. It will enable readers to gain insights on how vast volumes of data is simplified and how MapReduce is used in real-life applications. 
+
+### Introduction to MapReduce in Hadoop
+MapReduce is a Hadoop framework for writing applications that can process vast amounts of data on large clusters. It can also be termed a programming model in which large datasets can be processed across computer clusters. This application allows data to be stored in a distributed form. It simplifies huge volumes of data and large scale computing. 
+
+There are two main tasks in MapReduce: map and reduce. The map task is done first, followed by the reduce task. In the map job, the input dataset is split into chunks. Map task processes these chunks in a parallel manner. The map outputs are used as inputs for the Reduce tasks. Reducers process the intermediate data from the maps into smaller tuples, which reduce the tasks. This is the final output of the framework.
+
+The MapReduce framework enhances the scheduling and monitoring of tasks. The failed tasks are re-executed by the framework. This framework can be used easily, even by programmers with little expertise in distributed processing. MapReduce can be implemented using various programming languages such as Java, Hive, Pig, Scala, and Python. 	
+ 
+### How MapReduce in Hadoop works
+An overview of MapReduce Architecture and MapReduce's phases will help us understand how MapReduce in Hadoop works. 
+
+#### MapReduce Architecture
+The following diagram shows a MapReduce architecture.
+
+![MapReduce Architecture](/engineering-education/understanding-mapreduce-in-hadoop/map-reduce-architecture.png)
+
+[Image Source: A4Academics](http://a4academics.com/images/hadoop/Hadoop-Mapreduce-Architecture.png)
+
+MapReduce architecture consists of various components. A brief description of these components can improve our understanding of how MapReduce works. 
+- **Job:** This is an execution of a reducer or mapper across a dataset.
+- **Task:** This is an execution of a reducer or mapper on a section of data

```

Comment:
Why is this distinction so important? The difference between a Job and a Task? If the only difference is job is for the entire dataset and the task is for a part of the dataset, why would it be named this way? What is the need? Elaborate on that!

Reply:
I have revised accordingly.